src/Lachain.Storage/Repositories/PoolRepository.cs

```diff
@@ -101,6 +101,25 @@ public bool ContainsTransactionByHash(UInt256 txHash)
             return raw != null;
         }
 
+        [MethodImpl(MethodImplOptions.Synchronized)]
+        public void AddAndRemoveTransaction(TransactionReceipt txToAdd, TransactionReceipt txToRemove)
+        {
+            /* write transaction to storage */
+            var batch = new RocksDbAtomicWrite(_rocksDbContext);
+            var prefixTx = EntryPrefix.TransactionByHash.BuildPrefix(txToRemove.Hash);
+            batch.Delete(prefixTx);
+            prefixTx = EntryPrefix.TransactionByHash.BuildPrefix(txToAdd.Hash);
+            batch.Put(prefixTx, txToAdd.ToByteArray());
```

Comment:
Why we are not writing details in storage like this `_rocksDbContext.Save(EntryPrefix.TransactionPool.BuildPrefix(), pool.TransactionHashListToByteArray());`? 

Earlier code uses `Save` and `Delete` instead of `put` and `Delete`?

Reply:
RocksDbAtomicWrite uses Put instead of Save.
Using RocksDbAtomicWrite so that deleting old tx and writing new tx happens together when we call batch.Commit(). If the old tx is deleted but the new tx is not written or vice versa (for maybe power cut or node shut down), then it will be a problem.

Reply:
Yeah, Great handling